nixos/modules/services/web-apps/immich.nix

```diff
@@ -232,6 +232,10 @@ in
 
     services.postgresql = mkIf cfg.database.enable {
       enable = true;
+      package =
+        lib.warnIf (lib.versionOlder "1.135.3" pkgs.immich.version)
+          "nixos/immich: the services.postgresql.packge override may not be needed anymore."
+          pkgs.postgresql_16;
```

Comment:
This isn't a good idea because people might set a postgres version themselves. Maybe something between this priority and mkDefault?